picard/script.py

```diff
@@ -1201,3 +1201,22 @@ def func_slice(parser, multi, start_index, end_index, separator=MULTI_VALUED_JOI
         return separator.join(multi_var[start:end])
     except IndexError:
         return ''
+
+
+@script_function()
+def func_datetime(parser, format='%Y-%m-%d %H:%M:%S'):
+    """Return the current date and time as a string.
+
+    Arguments:
+        parser: The ScriptParser object used to parse the script.
+        format: A string or the ScriptVariable/Function that evaluates to the
+            string used to format the output.  Uses strftime() format.
+
+    Returns:
+        Returns he current date and time as a string.
+    """
+    import datetime
+    # Handle case where format evaluates to ''
+    if not format:
+        format = '%Y-%m-%d %H:%M:%S'
+    return datetime.datetime.now().strftime(format)
```

Comment:
I'd use `datetime.now().astimezone().strftime(format)`:

```python
>>> datetime.now().astimezone().strftime("%z")
'+0100'
>>> datetime.now().strftime("%z")
''
```

Not sure it works with Python3.5 though (@phw ?)

Reply:
Looking at the python docs it looks like it should.  I added it and we'll see if it passes the tests.